### PR TITLE
Tests for AffineSchemes

### PR DESCRIPTION
**Pull request by simonbrandhorst**
And some bugfixes.

### CODE REVIEW

File: experimental/Schemes/AffineSchemes.jl [hunk 1]
```diff
@@ -479,31 +481,31 @@ function issubset(
...
   UY = inverted_set(OO(Y))
-  if !issubset(UY, UX) 
+  if !issubset(UY, UX)
     # check whether the inverted elements in Y are units anyway
     for a in denominators(UY)
       is_unit(OO(X)(a)) || return false
     end
   end
-  return iszero(OO(Y)(modulus(OO(X))))
+  return true
```
Comment (afkafkafk13) [inline experimental/Schemes/AffineSchemes.jl]: I agree with the correctness of the change. For later reference, please comment in the code, why you do not need to worry about the modulus of X.

File: experimental/Schemes/AffineSchemes.jl [hunk 2]
```diff
@@ -643,6 +648,7 @@ function Base.intersect(
     X::AbsSpec{BRT, <:MPolyRing},
     Y::AbsSpec{BRT, <:MPolyQuoLocalizedRing}
   ) where {BRT}
+  @show "hi"
```
Comment (afkafkafk13) [inline experimental/Schemes/AffineSchemes.jl]: This was certainly debug output and should go away....

**Review by afkafkafk13 — COMMENTED** [review]
Please take care of the two minor comments.

Just curiosity: Concerning the tests, how much of the functionality of AffineSchemes.jl is now covered by tests (approximately)?

**Comment by thofma** [comment]
https://app.codecov.io/gh/oscar-system/Oscar.jl/blob/master/experimental/Schemes/AffineSchemes.jl

63% (before this PR)

Once the tests pass, there will be a `codecove/patch` item showing up in the lists of checks below. Then one can inspect how the coverage changed.

**Comment by fingolfin** [comment]
Once (if) tests pass, one also gets inline comments like this in the "Files" tab that indicates which lines modified by the PR are not covered. That can be useful: submit your code as a (draft) PR, see which parts are under tested, add tests, repeat.

Screenshot 2022-09-30 at 16 42 27

File: experimental/Schemes/AffineSchemes.jl [hunk 3]
```diff
@@ -110,7 +112,7 @@ base_ring_elem_type(X::AbsSpec) = base_ring_elem_type(typeof(X))
 On an affine scheme ``X/𝕜`` over ``𝕜`` this returns the ring `𝕜`.
 """
 function base_ring(X::AbsSpec{BRT, RT}) where {BRT, RT}
-  return base_ring(underlying_scheme(X))::BRT
+ return base_ring(underlying_scheme(X))::BRT
```
Comment (fingolfin) [inline experimental/Schemes/AffineSchemes.jl]: It's not a clear rule, but we generally use two spaces for indentation; some code also uses 4; much rarer is 3; but 1 is IMO really "not enough" ;-)
```suggestion
  return base_ring(underlying_scheme(X))::BRT
```

File: experimental/Schemes/AffineSchemes.jl (hunk 2 above)
Comment (fingolfin) [inline experimental/Schemes/AffineSchemes.jl]:
```suggestion
```

File: test/Schemes/AffineSchemes.jl [hunk 4]
```diff
@@ -34,6 +69,21 @@
...
+  @test issubset(EmptyScheme(QQ),A3)
+  @test issubset(EmptyScheme(QQ),U)
+  @test !issubset(U,EmptyScheme(QQ))
+  @test issubset(X,A3)
+  @test !issubset(A3, X)
+  @test issubset(A3,A3)
+  @test issubset(intersect(A3,A3), A3)
+  @test dim(A3) == 3
+  @test dim(U) == 3
+  @test dim(V) == 3
```
Comment (fingolfin) [inline test/Schemes/AffineSchemes.jl]: @simonbrandhorst This line throws an error (leading all tests to fail) because `V` is undefined

Comment (simonbrandhorst) [inline test/Schemes/AffineSchemes.jl]:
```suggestion
```

File: test/Schemes/AffineSchemes.jl [hunk 5]
```diff
@@ -44,3 +93,49 @@ end
...
+  S = powers_of_element(x)
+  Spec(R,S)
+  S = complement_of_ideal(I)
+  Spec(R,I)
+  X = subscheme(A3, I)
+  set_name!(X, "X")
+  @test iszero(OO(X)(f))
+  U = hypersurface_complement(A3, x)
+  @test issubset(intersect(A3,U),intersect(U,A3))
+  V  = PrincipalOpenSubset(U)
```
Comment (simonbrandhorst) [inline test/Schemes/AffineSchemes.jl:112]:
```suggestion
  V  = PrincipalOpenSubset(U)
  @test dim(V) == 3
```

File: experimental/Schemes/AffineSchemes.jl (hunk 1 above)
Comment (simonbrandhorst) [inline experimental/Schemes/AffineSchemes.jl]:
```suggestion
  return true  # Spec R/I[S^-1] is a closed subscheme of Spec R[S^-1]
```

**Comment by simonbrandhorst** [comment]
Thank you for your comments. I set to auto merge. Once there is a single positive review.

**Review by fingolfin — APPROVED** [review]

**Pull request merged**